st2reactor/st2reactor/container/partitioner.py

```diff
@@ -0,0 +1,165 @@
+            sensor_ref = sensor.get_reference()
+            if sensor_ref.ref in sensor_refs:
+                partition_members.append(sensor)
+
+        return partition_members
+
+    def get_required_sensor_refs(self):
+        return None
+
+
+class KVStorePartitioner(DefaultPartitioner):
+
+    def __init__(self, sensor_node_name):
+        super(KVStorePartitioner, self).__init__(sensor_node_name=sensor_node_name)
+        self._supported_sensor_refs = None
+
+    def is_sensor_supported(self, sensor_db):
+        """
+        """
+        return sensor_db.get_reference().ref in self._supported_sensor_refs
+
+    def get_required_sensor_refs(self):
+        partition_lookup_key = self._get_partition_lookup_key(self.sensor_node_name)
+
+        kvp = KeyValuePair.get_by_name(partition_lookup_key)
+        sensor_refs_str = kvp.value if kvp.value else ''
+        self._supported_sensor_refs = sets.Set([
+            sensor_ref.strip() for sensor_ref in sensor_refs_str.split(',')])
+        return self._supported_sensor_refs
+
+    def _get_partition_lookup_key(self, sensor_node_name):
+        return '{}.sensor_partition'.format(sensor_node_name)
+
+
+class FileBasedPartitioner(DefaultPartitioner):
+
+    def __init__(self, sensor_node_name, partition_file):
+        super(FileBasedPartitioner, self).__init__(sensor_node_name=sensor_node_name)
+        self.partition_file = partition_file
+        self._supported_sensor_refs = None
+
+    def is_sensor_supported(self, sensor_db):
+        """
+        All sensors are supported
+        """
+        return sensor_db.get_reference().ref in self._supported_sensor_refs and sensor_db.enabled
+
+    def get_required_sensor_refs(self):
+        with open(self.partition_file, 'r') as f:
+            partition_map = yaml.safe_load(f)
+            sensor_refs = partition_map.get(self.sensor_node_name, None)
+            if sensor_refs is None:
+                raise SensorPartitionMapMissingException('Sensor partition not found for %s in %s.',
+                                                         self.sensor_node_name, self.partition_file)
+            self._supported_sensor_refs = sets.Set(sensor_refs)
+            return self._supported_sensor_refs
+
+
+class SingleSensorProvider(object):
+
+    def __init__(self, sensor_ref):
+        self._sensor_ref = sensor_ref
+
+    def get_sensors(self):
+        sensor = SensorType.get_by_ref(self._sensor_ref)
+        if not sensor:
+            raise SensorNotFoundException('Sensor %s not found in db.' % self._sensor_ref)
+        return [sensor]
+
+    def is_sensor_supported(self, sensor_db):
+        """
+        No other sensor supported just the single sensor which was previously loaded.
+        """
+        return False
+
+
+PROVIDERS = {
+    DEFAULT_PARTITION_LOADER: DefaultPartitioner,
+    KVSTORE_PARTITION_LOADER: KVStorePartitioner,
+    FILE_PARTITION_LOADER: FileBasedPartitioner
+}
+
+
+def get_sensors_provider():
```

Comment:
It provides sensors by partitioning them so a provider. I guess I named it per the client expected interface as opposed to its own nature but I don't have a strong allegiance to either.